src/cpp/xmlparser/XMLParser.cpp

```diff
@@ -1462,6 +1478,160 @@ XMLP_ret XMLParser::parseDynamicTypes(tinyxml2::XMLElement* p_root)
     return parseXMLTypes(p_root);
 }
 
+XMLP_ret XMLParser::parseLogConfig(tinyxml2::XMLElement* p_root)
+{
+    /*
+    <xs:element name="log">
+      <xs:complexType>
+        <xs:boolean name="use_default"/>
+        <xs:sequence>
+          <xs:element maxOccurs="consumer">
+            <xs:complexType>
+              <xs:element name="class" type="string" minOccurs="1" maxOccurs="1"/>
+              <xs:sequence>
+                <xs:element name="propertyType"/>
+              </xs:sequence>
+            </xs:complexType>
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+    */
+
+    XMLP_ret ret = XMLP_ret::XML_OK;
+    tinyxml2::XMLElement *p_aux0 = nullptr;
+    p_aux0 = p_root->FirstChildElement(LOG);
+    tinyxml2::XMLElement* p_element = nullptr;
+
+    if (p_aux0 != nullptr)
+    {
+        p_element = p_aux0->FirstChildElement(USE_DEFAULT);
```

Comment:
Too much splitting between `p_aux0` and `p_root`. Join all code getting the XMLElement in an auxiliary variable and using it. Also be more restrictive. Right now this code permits any other not support tag.

Reply:
I have rewritten this section to validate all the tags and I have removed the split version unifying the management.